[MTIA] Support module.mtia()

egienvalue opened the pull request:
Summary: Following other device backends' implementation to support module.mtia() API.

Test Plan: OSS and Internal CIs.

Differential Revision: D60076584

mikaylagawarecki commented on tools/autograd/templates/python_variable_methods.cpp:
```diff
@@ -1272,6 +1295,7 @@ PyMethodDef variable_methods[] = {
   {"copy_", castPyCFunctionWithKeywords(THPVariable_copy_), METH_VARARGS | METH_KEYWORDS, NULL},
   {"cpu", castPyCFunctionWithKeywords(THPVariable_cpu), METH_VARARGS | METH_KEYWORDS, NULL},
   {"cuda", castPyCFunctionWithKeywords(THPVariable_cuda), METH_VARARGS | METH_KEYWORDS, NULL},
+  {"mtia", castPyCFunctionWithKeywords(THPVariable_mtia), METH_VARARGS | METH_KEYWORDS, NULL},
```
If the tensor method is public, please add a docstring like here https://github.com/pytorch/pytorch/blob/main/torch/_tensor_docs.py?fbclid=IwZXh0bgNhZW0CMTEAAR04QgPYDYYAq_jkjBzER9Y0JvxdiFDd23kdPwgqmOyvTkJUlcBW1Z50J14_aem_ZT-WF_2poV4qyHBFpTMz9Q#L1391-L1411

mikaylagawarecki reviewed (commented):
overrides failure needs an entry here https://github.com/pytorch/pytorch/blob/980bb543616ce971b4472f51f49e004fb33a6b7b/torch/overrides.py#L1406

egienvalue committed:
[MTIA] Support module.mtia() (#131499)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/131499

Following other device backends' implementation to support module.mtia() API.

Test Plan: OSS and Internal CIs.

Differential Revision: D60076584

mikaylagawarecki reviewed (approved):
Thanks
